src/Microsoft.AspNetCore.SignalR.Core/HubEndPoint.cs

```diff
@@ -32,16 +34,31 @@ public class HubEndPoint<THub> where THub : Hub
             _lifetimeManager = lifetimeManager;
             _loggerFactory = loggerFactory;
             _hubOptions = hubOptions.Value;
+            _globalHubOptions = globalHubOptions.Value;
             _logger = loggerFactory.CreateLogger<HubEndPoint<THub>>();
             _userIdProvider = userIdProvider;
             _dispatcher = dispatcher;
         }
 
         public async Task OnConnectedAsync(ConnectionContext connection)
         {
-            var connectionContext = new HubConnectionContext(connection, _hubOptions.KeepAliveInterval, _loggerFactory);
+            // We check to see if HubOptions<THub> are set because those take precedence over global hub options.
+            var keepAlive = _hubOptions.KeepAliveInterval ?? _globalHubOptions.KeepAliveInterval;
+            var negotiateTimeout = _hubOptions.NegotiateTimeout ?? _globalHubOptions.NegotiateTimeout;
+            var supportedProtocols = _hubOptions.SupportedProtocols;
+            if (supportedProtocols == null && _globalHubOptions.SupportedProtocols != null)
+            {
+                supportedProtocols = _globalHubOptions.SupportedProtocols;
+            }
+
+            if (supportedProtocols != null && supportedProtocols.Count == 0)
+            {
+                throw new InvalidOperationException("There are no supported protocols");
+            }
+
+            var connectionContext = new HubConnectionContext(connection, keepAlive.Value, _loggerFactory);
 
-            if (!await connectionContext.NegotiateAsync(_hubOptions.NegotiateTimeout, _protocolResolver, _userIdProvider))
+            if (!await connectionContext.NegotiateAsync(negotiateTimeout.Value, supportedProtocols, _protocolResolver, _userIdProvider))
```

Comment:
This can throw if negotiateTimeout is null...

Reply:
It shouldn't be null since it gets set in the HubOptionsSetup

Reply:
What if I explicitly set it to null?